app/models/trip.rb

```diff
@@ -0,0 +1,132 @@
+class Trip < ActiveRecord::Base
+  validates :duration,
+            :start_date,
+            :start_station_name,
+            :start_station_id,
+            :end_date,
+            :end_station_name,
+            :end_station_id,
+            :bike_id,
+            :subscription_type,
+            :zip_code, presence: true
+
+  belongs_to  :starting_station, class_name: :Station, foreign_key: :start_station_id
+  belongs_to  :end_station, class_name: :Station, foreign_key: :end_station_id
+  has_one :condition, foreign_key: :date, primary_key: :start_date
+
+  def self.average_duration
+    average(:duration)
+  end
+
+  def self.longest_ride
+    maximum(:duration)
+  end
+
+  def self.shortest_ride
+    minimum(:duration)
+  end
+
+  def self.most_popular_starting_station
+    trip_uses = group(:start_station_id).count
+    highest_use = group(:start_station_id).count.values.max
+    station_id = trip_uses.key(highest_use)
+    Station.find(station_id)
+  end
+
+  def self.most_popular_ending_station
+    trip_uses = group(:end_station_id).count
+    highest_use = group(:end_station_id).count.values.max
+    station_id = trip_uses.key(highest_use)
+    Station.find(station_id)
+  end
+
+  def self.month_by_month_breakdown_2013
+    year_2013 = where('extract(year FROM start_date)= ?', 2013)
+    sorted_by_month = year_2013.group("date_trunc('month', start_date)").count
+    convert_time_date_to_month(sorted_by_month)
+  end
+
+  def self.month_by_month_breakdown_2014
+    year_2014 = where('extract(year FROM start_date) = ?', 2014)
+    sorted_by_month = year_2014.group("date_trunc('month', start_date)").count
+    convert_time_date_to_month(sorted_by_month)
+  end
+
+  def self.month_by_month_breakdown_2015
```

Comment:
same with these monthly breakdown methods. the only difference is the initial query. can we perhaps have these methods take an argument (the year) and re-use this code?